server.go

```diff
@@ -2529,6 +2424,88 @@ func newServer(listenAddrs []string, db database.DB, chainParams *chaincfg.Param
 	return &s, nil
 }
 
+// initListeners initializes the configured net listeners and adds any bound
+// addresses to the address manager. Returns the listeners and a NAT interface,
+// which is non-nil if UPnP is in use.
+func initListeners(amgr *addrmgr.AddrManager, listenAddrs []string, services wire.ServiceFlag) ([]net.Listener, NAT, error) {
+	var nat NAT
+	discover := true
```

Comment:
I know it was already this way, but it seems to me like this would be cleaner as:

```Go
discover := len(cfg.ExternalIPs) == 0
if !discover {
	defaultPort, err := ...
	...
} else {
	...
}
```